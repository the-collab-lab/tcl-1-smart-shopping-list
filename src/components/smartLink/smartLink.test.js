import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import SmartLink from './index';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <SmartLink className="test-class">this is a smart link</SmartLink>
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('snapshot on load', () => {
  const tree = TestRenderer.create(
    <BrowserRouter>
      <SmartLink className="test-class">this is a smart link</SmartLink>
    </BrowserRouter>
  );
  expect(tree.toJSON()).toMatchSnapshot();
});

describe('test content values', () => {
  it('content is type string', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <SmartLink className="test-class">this is a smart link</SmartLink>
      </BrowserRouter>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('content is an html element', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <SmartLink className="test-class">
          <h1>this is an h1 element</h1>
        </SmartLink>
      </BrowserRouter>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('content is a React element', () => {
    const FakeComponent = () => <span>this is a fake component</span>;

    const tree = TestRenderer.create(
      <BrowserRouter>
        <SmartLink className="test-class">
          <FakeComponent />
        </SmartLink>
      </BrowserRouter>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('test routeTo values', () => {
  it('routeTo is set to known route', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <SmartLink className="test-class" routeTo="/add-item">
          this is a smart link
        </SmartLink>
      </BrowserRouter>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('routeTo is set to unknown route', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <SmartLink className="test-class" routeTo="/fake-link">
          this is a smart link
        </SmartLink>
      </BrowserRouter>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('routeTo is set to an external URL', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <SmartLink className="test-class" routeTo="http://google.com">
          this is a smart link
        </SmartLink>
      </BrowserRouter>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('routeTo is not set', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <SmartLink className="test-class">this is a smart link</SmartLink>
      </BrowserRouter>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('test visualState values', () => {
  it('visualState set to default', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <SmartLink className="test-class" visualState="default">
          this is a smart link
        </SmartLink>
      </BrowserRouter>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('visualState set to hidden', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <SmartLink className="test-class" visualState="hidden">
          this is a smart link
        </SmartLink>
      </BrowserRouter>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('visualState set to disabled', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <SmartLink className="test-class" visualState="disabled">
          this is a smart link
        </SmartLink>
      </BrowserRouter>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('visualState is not set', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <SmartLink className="test-class">this is a smart link</SmartLink>
      </BrowserRouter>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
