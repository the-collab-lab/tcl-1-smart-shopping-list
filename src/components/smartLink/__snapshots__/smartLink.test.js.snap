// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot on load 1`] = `
<a
  className="css-gq9b4g"
  href="/"
  onClick={[Function]}
  visualstate="default"
>
  this is a smart link
</a>
`;

exports[`test content values content is a React element 1`] = `
<a
  className="css-gq9b4g"
  href="/"
  onClick={[Function]}
  visualstate="default"
>
  <span>
    this is a fake component
  </span>
</a>
`;

exports[`test content values content is an html element 1`] = `
<a
  className="css-gq9b4g"
  href="/"
  onClick={[Function]}
  visualstate="default"
>
  <h1>
    this is an h1 element
  </h1>
</a>
`;

exports[`test content values content is type string 1`] = `
<a
  className="css-gq9b4g"
  href="/"
  onClick={[Function]}
  visualstate="default"
>
  this is a smart link
</a>
`;

exports[`test routeTo values routeTo is not set 1`] = `
<a
  className="css-gq9b4g"
  href="/"
  onClick={[Function]}
  visualstate="default"
>
  this is a smart link
</a>
`;

exports[`test routeTo values routeTo is set to an external URL 1`] = `
<a
  href="http://google.com"
  target="_blank"
  visualstate="default"
>
  this is a smart link
</a>
`;

exports[`test routeTo values routeTo is set to known route 1`] = `
<a
  className="css-gq9b4g"
  href="/add-item"
  onClick={[Function]}
  visualstate="default"
>
  this is a smart link
</a>
`;

exports[`test routeTo values routeTo is set to unknown route 1`] = `
<a
  className="css-gq9b4g"
  href="/fake-link"
  onClick={[Function]}
  visualstate="default"
>
  this is a smart link
</a>
`;

exports[`test visualState values visualState is not set 1`] = `
<a
  className="css-gq9b4g"
  href="/"
  onClick={[Function]}
  visualstate="default"
>
  this is a smart link
</a>
`;

exports[`test visualState values visualState set to default 1`] = `
<a
  className="css-gq9b4g"
  href="/"
  onClick={[Function]}
  visualstate="default"
>
  this is a smart link
</a>
`;

exports[`test visualState values visualState set to disabled 1`] = `
<a
  className="css-1k806u0"
  href="/"
  onClick={[Function]}
  visualstate="disabled"
>
  this is a smart link
</a>
`;

exports[`test visualState values visualState set to hidden 1`] = `
<a
  className="css-nuywz6"
  href="/"
  onClick={[Function]}
  visualstate="hidden"
>
  this is a smart link
</a>
`;
